### PR TITLE
feat: Adding secrets documentation

### PR DESCRIPTION
**Pull request by yhakbar**
Adds documentation to help customers make good decisions about secrets stored as part of Pipelines usage.

### CODE REVIEW

File: _docs-sources/pipelines/security/secrets.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+Populating this `var.cloudflare_api_token` for the provider requires having a `variable "cloudflare_api_token" {}`
+block in a `.tf` file that is checked into the repository, and setting the `TF_VAR_cloudflare_api_token` environment
+variable set to the value of the CloudFlare API token. The easiest way to do this is to leverage the `inputs` value
+in `terragrunt.hcl` files to set the value of the `cloudflare_api_token` variable to the value
+of the `CLOUDFLARE_API_TOKEN`
+
+```hcl
+inputs = {
+  cloudflare_api_token = "${run_cmd("--terragrunt-non-interactive", "./fetch-cloudflare-api-token.sh")}"
```
Comment (denis256) [inline _docs-sources/pipelines/security/secrets.md:107]:
```
ERRO[0000]   on /projects/gruntwork/terragrunt-tests/terragrunt.hcl line 2, in locals: 
ERRO[0000]    2:   api_token = run_cmd("--terragrunt-non-interactive", "./fetch-cloudflare-api-token.sh") 
ERRO[0000]                                              
ERRO[0000] Call to function "run_cmd" failed: exec: "--terragrunt-non-interactive": executable file not found in $PATH.
 
ERRO[0000] /projects/gruntwork/terragrunt-tests/terragrunt.hcl:2,15-23: Error in function call; Call to function "run_cmd" failed: exec: "--terragrunt-non-interactive": executable file not found in $PATH. 

```

Comment (denis256) [reply _docs-sources/pipelines/security/secrets.md:107]: works:
```
api_token = run_cmd("./fetch-cloudflare-api-token.sh", "--terragrunt-non-interactive")
```

Comment (yhakbar) [reply _docs-sources/pipelines/security/secrets.md]: Thanks, I was going for `--terragrunt-quiet`.

File: docs/pipelines/security/secrets.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+to improve the security posture of Pipelines, and are actively working on ways to reduce even this minimal requirement.
+
+## Authenticating With GitHub
+
+To authenticate with GitHub, Pipelines uses [Personal Access Tokens (PATs)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) to authenticate as GitHub Machine Users to interact with the GitHub API. You can learn more about how these tokens are created and managed in the [Machine Users](./machine-users.mdx) documentation.
+
+## Authenticating With AWS
+
+At a minimum, Pipelines also needs to authenticate with AWS. It does not do so with long lived credentials, however.
+Instead, it leverages [OIDC to authenticate with AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) as a particular git reference on a particular repository to assume
```
Comment (ZachGoldberg) [inline docs/pipelines/security/secrets.md:22]:
```suggestion
Instead, it leverages [OIDC to authenticate with AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services).  OIDC allows for an authenticated relationship to be defined between a particular git reference in a particular repository with a particular role in AWS, allowing pipelines to assume a role simply by virtue of where that pipeline is run. 
```

File: docs/pipelines/security/secrets.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+a role in AWS that trusts role assumptions from that context.
```
Comment (ZachGoldberg) [inline docs/pipelines/security/secrets.md:23]:
```suggestion
```

File: docs/pipelines/security/secrets.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+of the `CLOUDFLARE_API_TOKEN`
+
+```hcl
+inputs = {
+  cloudflare_api_token = "${run_cmd("--terragrunt-non-interactive", "./fetch-cloudflare-api-token.sh")}"
+}
+```
+
+:::note
+Here, `fetch-cloudflare-api-token.sh` is a script that fetches the CloudFlare API token from a secret store and prints it to stdout.
```
Comment (ZachGoldberg) [inline docs/pipelines/security/secrets.md:112]: I wonder if we might recommend a best practice, something like Murmur, to allow platform engineers to specify a list of secret locations they want in source control, and pipelines auto-fetches them and maps them to env.  Something like https://github.com/busser/murmur.

**Review by ZachGoldberg — APPROVED** [review]
Overall LGTM.  I think this document highlights an opportunity for pipelines to make this simpler, rather than a bunch of shell scripts to call a secret manager, we could offer a first class mechanism to define secret locations and envvar destinations.  Not a blocker, but something to consider.

**Pull request merged**